### PR TITLE
chore: Add HashMap docs

### PR DESCRIPTION
**Pull request by jfecher**
# Description

## Summary\*

Adds docs for https://github.com/noir-lang/noir/pull/4242

I've also edited the interface for `entries`, `keys`, and `values` to return `BoundedVec`s instead of arrays with optional elements. This is much more natural I think. I originally intended this for a separate PR, but I didn't want to write documentation for the old version of these functions only to immediately have to edit them afterward.

## Documentation\*

Check one:
- [ ] No documentation needed.
- [x] Documentation included in this PR.
- [ ] **[Exceptional Case]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: test_programs/execution_success/hashmap/src/main.nr [hunk 1]
```diff
@@ -154,9 +156,9 @@ fn test_iterators() {
     hashmap.insert(5, 7);
     hashmap.insert(11, 13);
 
-    let keys: [K; 3] = cut(hashmap.keys()).map(|k: Option<K>| k.unwrap_unchecked()).sort_via(K_CMP);
-    let values: [V; 3] = cut(hashmap.values()).map(|v: Option<V>| v.unwrap_unchecked()).sort_via(V_CMP);
-    let entries: [(K, V); 3] = cut(hashmap.entries()).map(|e: Option<(K, V)>| e.unwrap_unchecked()).sort_via(KV_CMP);
+    let keys: [K; 3] = cut(hashmap.keys()).sort_via(K_CMP);
+    let values: [V; 3] = cut(hashmap.values()).sort_via(V_CMP);
+    let entries: [(K, V); 3] = cut(hashmap.entries()).sort_via(KV_CMP);
```
Comment (jfecher) [inline test_programs/execution_success/hashmap/src/main.nr:161]: entries, keys, & values return a `BoundedVec` now so `cut` can return non-optional values.

**Comment by jfecher** [comment]
Looks like using the snippets script messes up the formatting in the snippets: https://65e5eac2f2a1ab3dd25576d3--noir-docs.netlify.app/docs/dev/noir/standard_library/containers/hashmap

**Comment by TomAFrench** [comment]
Generally looks good from a skim over the code. I'll do a proper read tomorrow morning and merge.

**Comment by Savio-Sou** [comment]
> Looks like using the snippets script messes up the formatting in the snippets:

Let's open up another Issue to address that. (Also noticed the links to source code link to master instead of a specific commit, which in the `get` case the line numbers no longer highlight the right portion of code as the file was updated.)
image

**Review by Savio-Sou — APPROVED** [review]
Merging this in so that we can get it as a part of v0.25.0.

For any further amendments that we might later want to make, we can create separate PRs.

**Comment by jfecher** [comment]
@Savio-Sou 

> which in the get case the line numbers no longer highlight the right portion of code as the file was updated.)

The lines of code are correct in the PR, so they'll link to the correct place once merged at least. Going forward they can break when looking at old versions of the docs though if we don't update the line numbers on the snippets there.

**Pull request merged**